app/services/registration_transfer_service.rb

```diff
@@ -27,18 +19,56 @@ def find_transient_registration
   def find_user(email)
     return nil unless email.present?
 
-    ExternalUser.where(email: email).first
+    return unless ExternalUser.where(email: email).first
+
+    # If a matching user exists, transfer and return a success status
+    update_account_emails(email)
+    send_existing_account_confirmation_email
+
+    :success_existing_user
+  end
+
+  def create_new_user(email)
```

Comment:
I'd like to see this method name changed. `create_new_user` infers it'll create a one using the email address provided and return it, but in fact it'll go on and complete the transfer, then send a status result back. 
```suggestion
  def transfer_new_user(email)
```